x/merkledb/intermediate_node_db.go

```diff
@@ -108,17 +108,43 @@ func (db *intermediateNodeDB) Get(key path) (*node, error) {
 	}
 	db.metrics.IntermediateNodeCacheMiss()
 
-	prefixedKey := addPrefixToKey(db.bufferPool, intermediateNodePrefix, key.Bytes())
+	dbKey := db.constructDBKey(key)
 	db.metrics.DatabaseNodeRead()
-	nodeBytes, err := db.baseDB.Get(prefixedKey)
+	nodeBytes, err := db.baseDB.Get(dbKey)
 	if err != nil {
 		return nil, err
 	}
-	db.bufferPool.Put(prefixedKey)
+	db.bufferPool.Put(dbKey)
 
 	return parseNode(key, nodeBytes)
 }
 
+// constructDBKey creates a key that can be used in the base db
+// We need to be able to differentiate between two paths of equal byte length but different token length,
+// so it adds a padding that can be used to differentiate.
+// Additionally, it adds the prefix indicating it is part of the intermediateDB
+func (db *intermediateNodeDB) constructDBKey(key path) []byte {
```

Comment:
I think we should probably write a couple explicit tests for this function - but looks great.